exo-go/src/types/index.go

```diff
@@ -17,18 +17,36 @@ type DependencyConfig struct {
 	ServiceEnvironment    map[string]string `yaml:"service-environment,omitempty"`
 }
 
-// ServiceConfig represents the configuration of a service
+// ServiceConfig represents the configuration of a service as provided in
+// service.yml
 type ServiceConfig struct {
+	Type            string            `yaml:",omitempty"`
+	Description     string            `yaml:",omitempty"`
+	Author          string            `yaml:",omitempty"`
+	Setup           string            `yaml:",omitempty"`
+	Startup         map[string]string `yaml:",omitempty"`
+	Production      map[string]string `yaml:",omitempty"`
+	ServiceMessages `yaml:",omitempty"`
+}
+
+// ServiceData represents the service info as provided in application.yml
+type ServiceData struct {
 	Location    string `yaml:",omitempty"`
 	DockerImage string `yaml:"docker-image,omitempty"`
 	NameSpace   string `yaml:",omitempty"`
 	Silent      bool   `yaml:",omitempty"`
 }
 
+// ServiceMessages represents the messages that the service sends and receives
+type ServiceMessages struct {
+	Receives []string
+	Sends    []string
+}
+
 // Services represents the mapping of protection level to services
 type Services struct {
-	Public  map[string]ServiceConfig
-	Private map[string]ServiceConfig
+	Public  map[string]ServiceData
+	Private map[string]ServiceData
 }
 
 // AppConfig represents the configuration of an application
```

Comment:
Changes taken from: https://github.com/Originate/exosphere/pull/334, although I haven't extracted them into their separate files yet. I thought that would make this PR too big - but I'll make sure to use @cheahuychou's changes when we do merge